From 4f0d3095f295ac3cebc9afc4b4bcd81768bd395a Mon Sep 17 00:00:00 2001
From: lelign <ignatlelekin@gmail.com>
Date: Thu, 7 May 2026 16:46:26 +0300
Subject: [PATCH] 
Upstream-Status: Pending
 renamed_mtv_overlay_remove_int_to_void_and_deleted_THIS_MODULEfrom__mtv_overlay_init

---
 mtv-overlay.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mtv-overlay.c b/mtv-overlay.c
index 7e9b454..f0ac9c8 100644
--- a/mtv-overlay.c
+++ b/mtv-overlay.c
@@ -226,7 +226,9 @@ long mtv_ioctl(struct file *filp,
         return 0;
 }
 
-int mtv_overlay_remove(struct platform_device *pdev)
+//int mtv_overlay_remove(struct platform_device *pdev)
+// .remove = mtv_overlay_remove,
+void mtv_overlay_remove(struct platform_device *pdev)
 {
         struct board_info * board = platform_get_drvdata(pdev);
 
@@ -241,7 +243,7 @@ int mtv_overlay_remove(struct platform_device *pdev)
         iounmap(board->mem_reg);
         release_mem_region(board->mem1_start, board->mem1_size);
         device_destroy(board_info_grob.mtv_class, MKDEV(board_info_grob.mtv_major, board->index));
-        return 0;
+        // return 0;
 }
 
 MODULE_DEVICE_TABLE(of, mtv_overlay_of_match);
@@ -257,7 +259,8 @@ static struct platform_driver mtv_overlay_driver = {
 
 static int mtv_overlay_init(void)
 {
-        board_info_grob.mtv_class = class_create(THIS_MODULE, DRV_NAME);
+        //board_info_grob.mtv_class = class_create(THIS_MODULE, DRV_NAME);
+        board_info_grob.mtv_class = class_create(DRV_NAME);
         board_info_grob.mtv_major = register_chrdev(0, DRV_NAME, &mtv_fops);
 
         return platform_driver_probe(&mtv_overlay_driver, mtv_overlay_probe);
